Client/Frontend/Fakespot/Client/ProductAnalysisData.swift

```diff
@@ -5,12 +5,12 @@
 import Foundation
 
 struct ProductAnalysisData: Codable {
-    let productId: String
-    let grade: String
-    let adjustedRating: Double
-    let needsAnalysis: Bool
-    let analysisUrl: URL
-    let highlights: Highlights
+    let productId: String?
+    let grade: String?
+    let adjustedRating: Double?
+    let needsAnalysis: Bool?
+    let analysisUrl: URL?
+    let highlights: Highlights?
 
```

Comment:
The API is returning a nil for every field in case there's no information about the product. This is strange since I expected and error or a nil object all together to be thrown. For now I made all properties optional to avoid decoding errors.